astropy/io/ascii/src/tokenizer.h

```diff
@@ -13,6 +13,14 @@
 
 #ifdef _MSC_VER
 #define inline __inline
+    #ifndef NAN
+        static const unsigned long __nan[2] = {0xffffffff, 0x7fffffff};
```

Comment:
On second look, in fairness this only needs to work on Windows platforms and MS compilers, so maybe doesn't matter.